src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobScheduler.scala

```diff
@@ -507,21 +507,24 @@ class JobScheduler @Inject()(val scheduleHorizon: Period,
           (None, None)
         } else {
           val job = jobOption.get
-          if (nextDate.isAfter(now.minus(job.epsilon)) && nextDate.isBefore(now.plus(scheduleHorizon))) {
+          val scheduleWindowBegin = now.minus(job.epsilon)
+          val scheduleWindowEnd = now.plus(scheduleHorizon)
+          if (nextDate.isAfter(scheduleWindowBegin) && nextDate.isBefore(scheduleWindowEnd)) {
             log.info("Task ready for scheduling: %s".format(nextDate))
             //TODO(FL): Rethink passing the dispatch queue all the way down to the ScheduledTask.
             val task = new ScheduledTask(TaskUtils.getTaskId(job, nextDate), nextDate, job, taskManager)
-            return (Some(task), stream.tail())
+            return (Some(task), stream.tail)
           }
-          //The nextDate has passed already beyond epsilon.
-          //TODO(FL): Think about the semantics here and see if it always makes sense to skip ahead of missed schedules.
+          // Next instance is too far in the future
+          // Needs to be scheduled at a later time, after schedule horizon.
           if (!nextDate.isBefore(now)) {
```

Comment:
It was me erroneously adding this call before. It does not belong here, we are not skipping any instances in this branch of the code, this branch is when the next instance to be scheduled is in the future.